src/parser/metrics/Metric.ts

```diff
@@ -117,3 +117,19 @@ export class ParsedFile extends SourceFile {
 export function isParsedFile(file: SourceFile): file is ParsedFile {
     return (file as ParsedFile).language !== undefined && (file as ParsedFile).tree !== undefined;
 }
+
+export class ErrorFile extends SourceFile {
+    /**
+     * Error that occured while processing the file.
+     */
+    error: Error;
+
+    constructor(filePath: string, error: Error) {
+        super(filePath);
+        this.error = error;
+    }
+}
+
+export function isErrorFile(file: SourceFile): file is ErrorFile {
+    return (file as ErrorFile).error !== undefined;
+}
```

Comment:
use instanceof instead?
also for isParsedFile?
=> new issue?

Reply:
Well, as SourceFile is a class now, that should work. However, I still hesitate to use instanceof because Ruben mentioned an issue with the jest test framework that can break the `instanceof` operator. See also #20.